File: app/src/main/java/com/kakacat/minitool/todayinhistory/Contract.java
package com.kakacat.minitool.todayinhistory;

import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;
import com.kakacat.minitool.todayinhistory.model.Article;

import java.util.List;

public interface Contract {

    interface Presenter extends IPresenter {

        void refreshData();

        int getYear();

        int getMonth();

        void setMonth(int month);

        int getDay();

        void setDay(int day);

        List<Article> getArticleList();
    }

    interface View extends IView<Presenter> {
        void onUpdateDataCallBack(String result, boolean needRefresh);

        void showCalendarDialog();
    }
}

File: app/src/main/java/com/kakacat/minitool/todayinhistory/Contract.kt
package com.kakacat.minitool.todayinhistory

import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView
import com.kakacat.minitool.todayinhistory.model.Article

interface Contract {
    interface Presenter : IPresenter {
        fun refreshData()
        val year: Int
        var month: Int
        var day: Int
        val articleList: List<Article?>?
    }

    interface View : IView<Presenter?> {
        fun onUpdateDataCallBack(result: String?, needRefresh: Boolean = false)
        fun showCalendarDialog()
    }
}